test/runtime/Policies.go

```diff
@@ -486,6 +486,33 @@ var _ = Describe("RuntimeValidatedPolicies", func() {
 		connectivityTest(allRequests, helpers.App2, helpers.Httpd1, true)
 	})
 
+	It("Tests Endpoint Regeneration Occurs When Daemon Configuration Is Updated", func() {
+		httpd1DockerNetworking, err := vm.ContainerInspectNet(helpers.Httpd1)
+
+		By("Importing policy and waiting for revision to increase for endpoints")
+		_, err = vm.PolicyImportAndWait(vm.GetFullPath(policiesL7JSON), helpers.HelperTimeout)
+		Expect(err).ToNot(HaveOccurred(), "unable to import policy after timeout")
+
+		By("Trying to access %s:80/public from %s (should be allowed by policy)", helpers.Httpd1, helpers.App1)
+		res := vm.ContainerExec(helpers.App1, helpers.CurlFail("http://%s:80/public", fmt.Sprintf(httpd1DockerNetworking[helpers.IPv4])))
+		res.ExpectSuccess("unable to access %s:80/public from %s (should have worked)", helpers.Httpd1, helpers.App1)
+
+		By("Changing daemon configuration")
+		res = vm.ExecCilium("config Debug=true")
+		res.ExpectSuccess("unable to change daemon configuration")
+
+		res = vm.ExecCilium("config Debug=false")
+		res.ExpectSuccess("unable to change daemon configuration")
+
+		By("Waiting for endpoints to be ready after daemon configuration change")
+		areEndpointsReady := vm.WaitEndpointsReady()
```

Comment:
If all endpoints are ready will pass, and the regeneration maybe not happen. Any endpoint revsion or endpoint log to check? Maybe something like `SetAndWaitForEndpointConfiguration()`?

Reply:
Investigating this. If we update the daemon's configuration such that the daemon's configuration has changed,  (which is done on lines 502 and 503), the endpoints will be forcibly regenerated:

The following detects if changes to configuration have been made, and forcibly triggers policy updates for the endpoints:

https://github.com/cilium/cilium/blob/ac2be0922e11f454c8f84b585ea02c5b0018e26f/daemon/daemon.go#L1422-L1434 

This call to `TriggerPolicyUpdates` eventually goes to the following code, with `force=true`:

https://github.com/cilium/cilium/blob/659d5acfa8c54f1257d3d57ab8d5732109ee1cca/pkg/endpointmanager/manager.go#L256-L289

This ensures that a build is queued for the endpoint, and regeneration will occur eventually. This goes in line with what you are saying that the endpoints will be ready, but may not have regenerated yet by the time this call to `WaitEndpointsReady` is made.

I guess that could be possible since the above code block for `TriggerPolicyUpdates` launches all of the regenerations for endpoints asynchronously, and in the configuration update code above we don't do anything with the waitgroup returned by the call to `d.TriggerPolicyUpdates`. and since calling `d.TriggerPolicyUpdates` bumps the policy revision when provided with `true` for the `force` argument, we should indeed wait for a policy revision bump per-endpoint. You are correct. I'll add a policy-revision-bump check in the next upload. Good catch :)

Reply:
Fixed.